include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -885,7 +885,11 @@ def ExtractRefOp : MooreOp<"extract_ref"> {
   }];
 }
 
-def StructCreateOp : MooreOp<"struct_create", [SameOperandsAndResultType]> {
+def StructCreateOp : MooreOp<"struct_create", [
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+  OptionalTypesMatchWith<"initial value and variable types match",
+    "result", "initial", "cast<RefType>($_self).getNestedType()">
+]> {
```

Comment:
#7039 <-- At this PR, @fabianschuiki mentioned `DestructurableAllocationOpInterface` and `DestructurableAccessorOpInterface` interfaces. So I think maybe you can flag `moore.struct_create` with the former and the `moore.struct_extract` and `moore.struct_extract_ref` with the latter.

Reply:
Search `MemorySlotInterfaces.td` to view the details.

Reply:
Thanks for reminding, I will check it.